Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2113,7 +2113,13 @@ extension BrowserViewController: ContextMenuHelperDelegate {
             let copyAction = UIAlertAction(title: copyImageTitle, style: UIAlertActionStyle.Default) { (action: UIAlertAction) -> Void in
                 let pasteBoard = UIPasteboard.generalPasteboard()
                 pasteBoard.string = url.absoluteString
-                // TODO: put the actual image on the clipboard
+                // put the actual image on the clipboard
+                // do this be asyncronously just in case we're in a low bandwidth situation
+                // Is this neccessary, or is the image already sat in cache?
+                NSURLSession.sharedSession().dataTaskWithURL(url, completionHandler: { (data, _, error) -> Void in
```

Comment:
I wonder if we should use Alamofire instead of NSURLSession instead. I know we use it for other things in the app like the FaviconFetcher [1] because we setup an instance with our user agent.

[1] https://github.com/mozilla/firefox-ios/blob/fluffyemily/Bug-1212860-copy-image-data-to-pasteboard/Utils/FaviconFetcher.swift#L68.